src/deluge/gui/views/automation_view.cpp

```diff
@@ -435,7 +439,7 @@ void AutomationView::graphicsRoutine() {
 // used to return whether Automation View is in the AUTOMATION_ARRANGER_VIEW UI Type, AUTOMATION_INSTRUMENT_CLIP_VIEW or
 // AUTOMATION_AUDIO_CLIP_VIEW UI Type
 AutomationSubType AutomationView::getAutomationSubType() {
-	if (onArrangerView) {
+	if (currentSong->onAutomationArrangerView) {
```

Comment:
This is brittle - if a future change allows opening a clip without leaving automation view (like launchpad integration or something), then this will no longer work. This needs to be based on the automation view itself

All other checks on currentSong->onAutomationArrangerView need to be replaced by calls to this function